static/js/message_live_update.js

```diff
@@ -14,14 +14,26 @@ function rerender_messages_view() {
     }
 }
 
+function rerender_messages_view_for_user(user_id) {
+    for (const list of [message_lists.home, message_list.narrowed]) {
+        if (list === undefined) {
+            continue;
+        }
+        if (list.table_name !== undefined) {
+            const msgs = list.data.get_messages_sent_by_user(user_id);
+            list.view.rerender_messages(msgs);
+        }
+    }
+}
+
 export function update_stream_name(stream_id, new_name) {
     message_store.update_property("stream_name", new_name, {stream_id});
     rerender_messages_view();
 }
 
 export function update_user_full_name(user_id, full_name) {
     message_store.update_property("sender_full_name", full_name, {user_id});
-    rerender_messages_view();
+    rerender_messages_view_for_user(user_id);
```

Comment:
I think `update_avatar` needs the same treatment?

Reply:
I see, this is covered in the commit message.  My guess is that we basically need this line:

`message_container.small_avatar_url = people.small_avatar_url(message_container.msg);` 

to be moved into
`set_calculated_message_container_variables`

Probably this whole block should be moved:

```
            message_container.sender_is_bot = people.sender_is_bot(message_container.msg);               
            message_container.sender_is_guest = people.sender_is_guest(message_container.msg);           
                                                                                                         
            message_container.small_avatar_url = people.small_avatar_url(message_container.msg);         
            if (message_container.msg.stream) {                                                          
                message_container.background_color = stream_data.get_color(                              
                    message_container.msg.stream,                                                        
                );                                                                                       
            }                                                                                            
```

Reply:
That's where I struggled most of today, what I'm seeing on local is that the avatars do not change, on inspection the avatar url does not change, but if I open that url in a new tab I see the new image.